tests/scripts/check-generated-files.sh

```diff
@@ -22,8 +22,12 @@ EOF
     exit
 fi
 
-if [ -d library -a -d include -a -d tests ]; then :; else
-    echo "Must be run from Mbed TLS root" >&2
+IN_MBEDTLS=0
+if [ -d library -a -d include -a -d tests ]; then
+    IN_MBEDTLS=1
+elif [ -d core -a -d include -a -d tests ]; then :;
```

Comment:
Deciding between mbedtls and tf-psa-crypto from `library` vs `core` feels fragile. How about testing the presence of `include/mbedtls` vs `include/tf_psa_crypto`?

Reply:
Please do it like in all.sh in this PR to keep things consistent. `include/tf_psa_crypto` was introduced recently thus not used yet in the repo root detection checks. We could use it and in that case we would update all such checks in all *.sh files.

Reply:
Sure, will do. I did think that however I thought the way I was doing it here look more consistent within this particular script. Happy to change it though

Reply:
Should we introduce a `scripts/which_repo.sh`? Or a `scripts/repo_name.txt`?

For Python code there should be a function in `build_tree.py`.

Reply:
@gilles-peskine-arm I am just wondering why doing it this way seems more fragile than using the `include` directory as you suggest? It's not clear to me why one approach is more robust than the other.

Reply:
> Should we introduce a `scripts/which_repo.sh`? Or a `scripts/repo_name.txt`?
Why not, I am not really fluent in *.sh "inclusions" thus do not know to best practices there. 
 
> For Python code there should be a function in `build_tree.py`.
We have it.

Reply:
The mere existence of `core` vs `library` might become obsolete when we work on Mbed TLS 4.0, and might go wrong if you switch between the two products in the same working directory.

Come to think of it, `include/DIR` isn't that good either. We should use a regular file.

Or just insist on the existence of a data file. We can decide to introduce that (unlike third-party scripts).

Reply:
> Should we introduce a `scripts/which_repo.sh`? Or a `scripts/repo_name.txt`?
> 
> For Python code there should be a function in `build_tree.py`.

I would be in favour of adding a `scripts/which_repo.sh` to handle all the repo detection. I think it would be cleaner and more modular.

Reply:
> The mere existence of `core` vs `library` might become obsolete when we work on Mbed TLS 4.0, and might go wrong if you switch between the two products in the same working directory.
> 
> Come to think of it, `include/DIR` isn't that good either. We should use a regular file.
> 
> Or just insist on the existence of a data file. We can decide to introduce that (unlike third-party scripts).

Sounds good, thanks for the explanation.

Reply:
@gilles-peskine-arm By "use a regular file" do you mean we should simply have a file named `mbedtls` in the root directory of Mbed TLS and then the same idea of `TF-PSA-Crypto` and then we test for it's existence with the proposed `scripts/which-repo.sh` script?

Reply:
No, I thought each repository should have a file with the same name, and we'd use the content of that file to identify the repository.

Reply:
Cool ok. In this PR I think it is probably best if I do it the way @ronald-cron-arm suggests in his first comment and then I will raise a subsequent PR that implements this file idea and modifies how repo detection is done across both the repositories. Does that sound alright to everyone here?

Reply:
fine by me

Reply:
The standardisation of repository detection is now being tracked with issue #8524 .